dotCMS/src/main/java/com/dotcms/security/apps/AppsCacheImpl.java

```diff
@@ -0,0 +1,213 @@
+package com.dotcms.security.apps;
+
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.business.DotCacheAdministrator;
+import com.dotmarketing.business.DotCacheException;
+import com.dotmarketing.util.UtilMethods;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+
+/**
+ * Cache Admin allows management of Apps Cache.
+ * Here we have The yml App Descriptors Cache as well as the secret storage.
+ * The class is just a entry point to the cache administrator.
+ * You can store secrets organized by key and also keep a list of all the keys that currently exist in the secret store.
+ */
+public class AppsCacheImpl extends AppsCache {
+
+    private DotCacheAdministrator cache;
+
+    /**
+     * Default constructor.
+     */
+    public AppsCacheImpl() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [It is a good practice to call super() in a constructor](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5358201)